### PR TITLE
Fix browserdeviceprofile returning wrong h264 level for gen 1 & 2

### PR DESCRIPTION
**Pull request by YouKnowBlom**
Another temporary fix. We really need a custom browserdeviceprofile but I don't want to make this PR much larger

### CODE REVIEW

File: components/maincontroller.js [hunk 1]
```diff
@@ -644,6 +644,21 @@
 
         profile.TranscodingProfiles = profile.TranscodingProfiles.filter(item => item.Container !== "mkv");
 
+        //Another temporary solution until we can make a custom browserdeviceprofile
+        //1st and 2nd gen only support h264 4.1 but browserdeviceprofile is reporting 4.2
+        let context = cast.framework.CastReceiverContext.getInstance();
+        if (!context.canDisplayType('audio/mp4;codecs=avc1.64002A')) {
+            for (prof of profile.CodecProfiles) {
```
Comment (cvium) [inline components/maincontroller.js]: I wonder if `filter` would look nicer here.

**Comment by YouKnowBlom** [comment]
I'm not sure what happened there, somehow your comment disappeared @cvium 

I feel like using `filter` would be kinda hacky, since I wouldn't be using `filter` for its intended use

I threw this together with `filter` anyway
```js
profile.CodecProfiles.filter( (prof) => {
                if (prof.Codec && prof.Codec.indexOf("h264") != -1) {
                    prof.Conditions.filter( (condition) => {
                        if (condition.Property == "VideoLevel") {
                            condition.Value = (condition.Value > 41)? 41 : condition.Value;
                        }
                        return true;
                    });
                }
                return true;
            });
```

File: components/maincontroller.js [hunk 2]
```diff
@@ -644,6 +644,21 @@
...
+                if (prof.Codec && prof.Codec.indexOf("h264") != -1) {
+                    for (condition of prof.Conditions) {
+                        if (condition.Property == "VideoLevel") {
+                            condition.Value = (condition.Value > 41)? 41 : condition.Value;
+                        }
+                    }
+                }
+            }
+        }
```
Comment (cvium) [inline components/maincontroller.js]: I would write it like this I think
```suggestion
        if (context.canDisplayType('audio/mp4;codecs=avc1.64002A')) {
            return profile;
        }
        
        for (prof of profile.CodecProfiles) {
            if (prof.Codec && prof.Codec.indexOf("h264") == -1) {
                continue;
            }
            for (condition of prof.Conditions) {
                if (condition.Property == "VideoLevel" && condition.Value > 41) {
                    condition.Value = 41;
                }
            }
        }
```

Comment (YouKnowBlom) [reply components/maincontroller.js]: That's much cleaner, I'll go with that 👍

File: components/maincontroller.js [hunk 3]
```diff
@@ -644,6 +644,25 @@
 
         profile.TranscodingProfiles = profile.TranscodingProfiles.filter(item => item.Container !== "mkv");
 
+        //Another temporary solution until we can make a custom browserdeviceprofile
+        //1st and 2nd gen only support h264 4.1 but browserdeviceprofile is reporting 4.2
```
Comment (cvium) [inline components/maincontroller.js]:
```suggestion
        // TODO: Another temporary solution until we can make a custom browserdeviceprofile
        // 1st and 2nd gen only support h264 4.1 but browserdeviceprofile is reporting 4.2
```

**Review by cvium — APPROVED** [review]

**Pull request merged**